addons/sale_management/models/sale_order.py

```diff
@@ -118,6 +118,12 @@ def _onchange_sale_order_template_id(self):
 
         self.sale_order_option_ids = option_lines_data
 
+    @api.onchange('partner_id')
+    def _onchange_partner_id(self):
+        """Reload template if number of lines matches to update translations."""
+        if len(self.sale_order_template_id.sale_order_template_line_ids) == len(self.order_line):
+            self._onchange_sale_order_template_id()
```

Comment:
Mhm, I don't think it's enough to only check the number of lines, triggering the onchange will reset their whole content, losing the quantity, prices and so on if they were modified.

Reply:
Yeah, figured that might be an issue. Changed it to also compare `create_date` to `write_date`. This way only unsaved changes run risk of getting yeeted.

Reply:
Also, suggestion for master: adding `sale.order.template.line` as a related field to `sale.order.line`?
Would allow updating translations regardless of modifications.